google/cloud/storage/internal/grpc_object_request_parser.cc

```diff
@@ -360,6 +371,101 @@ ListObjectsResponse GrpcObjectRequestParser::FromProto(
   return result;
 }
 
+StatusOr<google::storage::v2::RewriteObjectRequest>
+GrpcObjectRequestParser::ToProto(RewriteObjectRequest const& request) {
+  google::storage::v2::RewriteObjectRequest result;
+  SetCommonParameters(result, request);
+  auto status = SetCommonObjectParameters(result, request);
+  if (!status.ok()) return status;
+
+  auto& destination = *result.mutable_destination();
+  destination.set_name(request.destination_object());
+  destination.set_bucket("projects/_/buckets/" + request.destination_bucket());
+  destination.set_kms_key(
+      request.GetOption<DestinationKmsKeyName>().value_or(""));
+  if (request.HasOption<WithObjectMetadata>()) {
+    // Only a few fields can be set as part of the metadata request.
+    auto m = request.GetOption<WithObjectMetadata>().value();
+    destination.set_storage_class(m.storage_class());
+    destination.set_content_encoding(m.content_encoding());
+    destination.set_content_disposition(m.content_disposition());
+    destination.set_cache_control(m.cache_control());
+    destination.set_content_language(m.content_language());
+    destination.set_content_type(m.content_type());
+    destination.set_temporary_hold(m.temporary_hold());
+    for (auto const& kv : m.metadata()) {
+      (*destination.mutable_metadata())[kv.first] = kv.second;
+    }
+    if (m.event_based_hold()) {
+      // The proto is an optional<bool>, avoid setting it to `false`, seems
+      // confusing.
```

Comment:
Ok so false and unset are not the same things:
https://github.com/googleapis/googleapis/blob/ce08228300f5bf6a0c7153176e3adac501eaa5a1/google/storage/v2/storage.proto#L1984-L1986

you could add like a
```
bool ObjectMetadata::use_parent_default_event_based_hold() const;
```

(not saying you should. I'm just doing my due diligence as a reviewer)

Reply:
There is an ongoing discussion about how to deal with these optionals.  The protos may need to change to deal with this.  Yes, it is ugly.